the-state-machine/typescript/lib/the-state-machine-stack.ts

```diff
@@ -16,16 +16,19 @@ export class TheStateMachineStack extends cdk.Stack {
     //The first thing we need to do is see if they are asking for pineapple on a pizza
     let pineappleCheckLambda = new lambda.Function(this, 'pineappleCheckLambdaHandler', {
       runtime: lambda.Runtime.NODEJS_12_X,    // execution environment
-      code: lambda.Code.fromAsset('lambda-fns'),     // code loaded from the "lambda" directory
+      code: lambda.Code.fromAsset('lambda-fns'),     // code loaded from the "lambda-fns" directory
       handler: 'orderPizza.handler'           // file is "orderPizza", function is "handler"
     });
 
     // Step functions are built up of steps, we need to define our first step
-    const orderPizza = new sfn.Task(this, 'Order Pizza Job', {
-      task: new tasks.InvokeFunction(pineappleCheckLambda),
+    // Refactored code because sfn.Task is Depecrated right now. Using StepFunctionsTaks.LambdaInvoke instead
+    // https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.Task.html
+    const orderPizza = new tasks.LambdaInvoke(this, "Order Pizza Job", {
```

Comment:
This piece of code was obsolete (https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions.Task.html). I changed it to the new implementation using service as invocation (https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-stepfunctions-tasks.LambdaInvoke.html)